diff --git a/Makefile b/Makefile
index 3135d70..a2bf4b2 100644
--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ ifneq ($(BUILD_SHARED),1)
 endif
 endif

-INSTALL_TARGETS += $(bindir)/cbordump
+#INSTALL_TARGETS += $(bindir)/cbordump
 ifeq ($(BUILD_SHARED),1)
 BINLIBRARY=lib/libtinycbor.so
 INSTALL_TARGETS += $(libdir)/libtinycbor.so.$(VERSION)
diff --git a/src/open_memstream.c b/src/open_memstream.c
index 3365378..ba955d7 100644
--- a/src/open_memstream.c
+++ b/src/open_memstream.c
@@ -101,6 +101,8 @@ FILE *open_memstream(char **bufptr, size_t *lenptr)

 #ifdef __APPLE__
     return funopen(b, NULL, write_to_buffer, NULL, close_buffer);
+#elif __ANDROID__
+    abort();
 #elif __linux__
     static const cookie_io_functions_t vtable = {
         NULL,
